controllers/replica_set_port_manager.go

```diff
@@ -0,0 +1,109 @@
+package controllers
```

Comment:
> This probably should be in pkg/ which would allow it to be used in the enterprise operator

Indeed 

I wouldn't mind putting this under `automationconfig` package for now -- since it computes the ports for AC.

Reply:
@irajdeep We cannot move it easily into automationconfig pkg because it's using pod state logic from agent package. And we have a nasty import cycle there.
But I think it fits in agent package as it's logic is also related to mongodb-agent limitation around changing port one by one. Also core logic uses routine for checking goal states and that is also in agent pkg.
Moved to agent for now. Let me know if it's ok.

Reply:
> Moved to agent for now.

Sounds good.

Reply:
Done